### PR TITLE
Refs #30686 -- Fixed text truncation for negative or zero lengths.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

File: tests/utils_tests/test_text.py [hunk 1]
```diff
@@ -123,6 +123,10 @@ def test_truncate_chars_html(self):
             "…",
             truncator.chars(1, html=True),
         )
+        self.assertEqual(
+            "",
+            truncator.chars(0, html=True),
+        )
```
Comment (adamchainz) [inline tests/utils_tests/test_text.py:129]:
```suggestion
        self.assertEqual(
            "",
            truncator.chars(0, html=True),
        )
        self.assertEqual(
            "",
            truncator.chars(-1, html=True),
        )
```

File: tests/utils_tests/test_text.py [hunk 2]
```diff
@@ -206,6 +210,7 @@ def test_truncate_words(self):
             lazystr("The quick brown fox jumped over the lazy dog.")
         )
         self.assertEqual("The quick brown fox…", truncator.words(4))
+        self.assertEqual("", truncator.words(0))
```
Comment (adamchainz) [inline tests/utils_tests/test_text.py:213]:
```suggestion
        self.assertEqual("", truncator.words(0))
        self.assertEqual("", truncator.words(-1))
```

**Review by adamchainz — CHANGES_REQUESTED** [review]
Add negative tests for completeness?

**Comment by felixxm** [comment]
> Add negative tests for completeness?

Added.

**Pull request merged**